pkg/app/piped/notifier/slack.go

```diff
@@ -51,14 +53,30 @@ type slack struct {
 }
 
 func newSlackSender(name string, cfg config.NotificationReceiverSlack, webURL string, logger *zap.Logger) *slack {
+	var oauthtoken string
+	if cfg.OAuthTokenData != "" {
+		a, err := base64.StdEncoding.DecodeString(cfg.OAuthTokenData)
+		if err != nil {
+			logger.Error(fmt.Sprintf("failed to decode the oauth token data: %v", err))
+			return nil
+		}
+		oauthtoken = string(a)
+	}
+	if cfg.OAuthTokenFile != "" {
+		a, err := os.ReadFile(cfg.OAuthTokenFile)
+		if err != nil {
+			logger.Error(fmt.Sprintf("failed to read the oauth token file: %v", err))
+		}
+		oauthtoken = string(a)
+	}
```

Comment:
I think that isn't good. Because the caller of newSlackSender doesn't check if `*slack` is nil. So, if err isn't nil, it may be panic caused by nil pointer reference. In additional, the caller of top level function checks if err isn't nil. Therefore, I think we should change a signature of newSlackSender. I propose new signature `func newSlackSender(name string, cfg config.NotificationReceiverSlack, webURL string, logger *zap.Logger) (*slack, error)`.

Reply:
got your point on this, I already implemented that